src/components/Timeline/src/Step.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState, KeyboardEventHandler } from 'react';
+import MaterialStep from '@material-ui/core/Step';
+
+import { ChevronDownIcon, ChevronUpIcon } from '@gemeente-denhaag/icons';
+
+import { StepContent, StepLabel } from '.';
+import { stepClasses, stepCollapseIconClasses } from './bem-mapping';
+import { TimelineComponent } from './types';
+
+export interface StepProps extends TimelineComponent {
+  /**
+   * Describes the information of the step.
+   */
+  label: string;
+  /**
+   * Additional information about the step.
+   */
+  description?: string;
+}
+
+/**
+ * Represents a single Step in a Stepper component.
+ * @param props The properties of a Step component.
+ * @constructor Constructs an instance of Step.
+ */
+export const Step: React.FC<StepProps> = ({ active = false, tabIndex = 0, ...props }: StepProps) => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const hasDescription = props.description !== undefined;
+
+  if (active && props.disabled) {
+    throw new Error('Step cannot be active and disabled at the same time.');
+  }
+
+  const toggle = () => {
+    if (hasDescription && props.completed) {
+      setIsExpanded((prevState) => !prevState);
+    }
+  };
+
+  // If active has been updated, this indicates that the user is on the next
+  // step, then we want to collapse the StepContent
+  useEffect(() => {
+    setIsExpanded(false);
+  }, [active]);
+
+  // Toggle when element is focussed and 'Enter' is pressed.
+  const handleKeyDown: KeyboardEventHandler<HTMLDivElement> = (event) => {
+    if (event.key === 'Enter') {
+      toggle();
+    }
+  };
+
+  let descriptionElement: string | React.ReactElement = '';
+  if (hasDescription) {
+    descriptionElement = <StepContent>{props.description}</StepContent>;
+  }
+
+  let iconElement: string | React.ReactElement = '';
+  if (hasDescription && props.completed) {
+    if (isExpanded) {
+      iconElement = <ChevronUpIcon classes={stepCollapseIconClasses} />;
+    } else {
+      iconElement = <ChevronDownIcon classes={stepCollapseIconClasses} />;
+    }
+  }
+
+  return (
+    <MaterialStep
+      active={active}
+      classes={stepClasses}
+      expanded={isExpanded}
+      onClick={toggle}
+      onKeyDown={handleKeyDown}
+      tabIndex={tabIndex}
+      {...props}
```

Comment:
I believe our general way of working is to not include `{...props}` and instead individually call every prop to disallow the end-user to adjust props of the Material-UI component which we do not want to be adjustable.

Reply:
same story as https://github.com/Gemeente-DenHaag/denhaag-component-library/pull/221#discussion_r650394921

Reply:
When doing this, do not forget to also add the properties specified in `BaseProps` in the component

Reply:
I've added `id` to every component